### ISSUE TITLE
Layer styling docker doesn't accept decimal points in range inputs

### ISSUE THREAD

**Issue by lnicola** [qgis/QGIS#50861]
### What is the bug or the crash?

You can't type decimal points in the min/max input boxes, and the cursor moves around randomly.

### Steps to reproduce the issue

1. Load a float raster
2. Open Layer styling
3. Set it to Singleband grayscale and try to input a decimal point

### Versions

3.26. I'll update tomorrow.

### Supported QGIS version

- [X] I'm running a supported QGIS version according to the roadmap.

### New profile

- [ ] I tried with a new QGIS profile

**Comment by nicogodet** [comment]
> 3.26. I'll update tomorrow.

I were able to confirm on 3.28.0 when asked on IRC

**Comment by lnicola** [comment]
Did you test with a floating-point file? I'll give it a try when I find a 3.30 build.

**Comment by lnicola** [comment]
No, still happens on 3.30.1. When testing, make sure to use a `Float32` raster and to open the Styling Docker (<kbd>F7</kbd>), and not the layer properties.

**Comment by janzandr** [comment]
Ah ok, with "Docker" you mean the Layer Styling Panel.
Confirmed, I can reproduce the problem when "Live update" option is activated.
A workaround is to type your value without decimal point first, e.g. 123, and place the point afterwards, e.g. 1.23
